src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,174 @@
+/** @jsx h */
+
+import { Component, h } from 'preact';
+import cx from 'classnames';
+
+import PaginationLink from './PaginationLink';
+import { isSpecialClick } from '../../lib/utils';
+import {
+  PaginationCSSClasses,
+  PaginationTemplates,
+} from '../../widgets/pagination/pagination';
+
+export type PaginationComponentCSSClasses = {
+  [TClassName in keyof PaginationCSSClasses]: string;
+};
+
+export type PaginationProps = {
+  createURL(value: number): string;
+  cssClasses: PaginationComponentCSSClasses;
+  currentPage: number;
+  templates: PaginationTemplates;
+  nbPages?: number;
+  pages?: number[];
+  isFirstPage: boolean;
+  isLastPage: boolean;
+  setCurrentPage(value: number): void;
+  showFirst?: boolean;
+  showLast?: boolean;
+  showPrevious?: boolean;
+  showNext?: boolean;
+};
+
+type PageLink = {
+  label: string;
+  ariaLabel: string;
+  pageNumber: number;
+  additionalClassName: string | null;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  createURL(value: number): string;
+};
+
+const defaultProps = {
+  currentPage: 0,
+  nbPages: 0,
+  pages: [],
+};
+
+class Pagination extends Component<PaginationProps> {
+  public static defaultProps = defaultProps;
+
+  private pageLink({
+    label,
+    ariaLabel,
+    pageNumber,
+    additionalClassName = null,
+    isDisabled = false,
+    isSelected = false,
+    createURL,
+  }: PageLink) {
+    const cssClasses = {
+      item: cx(this.props.cssClasses.item, additionalClassName),
+      link: this.props.cssClasses.link,
+    };
+
+    if (isDisabled) {
+      cssClasses.item = cx(cssClasses.item, this.props.cssClasses.disabledItem);
+    } else if (isSelected) {
+      cssClasses.item = cx(cssClasses.item, this.props.cssClasses.selectedItem);
+    }
+
+    const url = !isDisabled ? createURL(pageNumber) : '#';
```

Comment:
is createURL no longer optional?

Reply:
Oops I mixed both types, it should be optional in the `Pagination` but not in `pageLink`, right?

Reply:
I checked and it's actually defined in all cases, so the check that was here before indeed isn't needed